src/components/IstioWizards/RequestRouting/RuleBuilder.tsx

```diff
@@ -84,7 +84,11 @@ class RuleBuilder extends React.Component<Props, State> {
         <Tabs isFilled={true} activeKey={this.state.ruleTabKey} onSelect={this.ruleHandleTabClick}>
           <Tab eventKey={0} title={'Request Matching'}>
             <div style={{ marginTop: '20px' }}>
-              <MatchBuilder {...this.props} />
+              <MatchBuilder
+                {...this.props}
+                matchOptions={[HEADERS, URI, SCHEME, METHOD, AUTHORITY]}
```

Comment:
I understand that the motivation of this change is to reuse the MatchBuilder in a Iter8 form.
I'd suggest that if you need to use these optional matchOptions/opOptions you could make these parameters optional, but leave the default behaviour inside the MatchBuilder logic.
Perhaps for "matchOptions" is easy, but the conditional array filled in the "opOptions" is something that belongs to the MatchBuilder logic, I think.

So, I'd suggest that when matchOptions/opOptions are present MatchBuilder can use them, but if not, it can get the default used for Kiali Wizards.

I think with that minimal change perhaps can be cleaner.

Also, typically this is an auxiliar component for the RequestRouting scenario, that's reason that this class is located in a deep path like "src/components/IstioWizards/RequestRouting/RuleBuilder.tsx", if you consider better, you can also clone the RuleBuilder/MatchBuilder and put a version of that clase under Iter8 logic and add that logic there.

In some cases some duplication sounds cleaner if justified, instead to make an aux component to work for both scenarios.

I'm fine as well with that strategy too if that simplifies your work.

Also, thinking in the future, cloning this aux components may give you more freedon to change them if the current logic can be maintained for both Iter8 / Istio needs.

Let me know what do you think.

@Alan-Cha Thanks for your PR and sorry for my delay in the review, past week was the IstioCon and most of the team was a little bit busy than usual.

Reply:
@lucasponce Both options, creating a default behavior for `MatchBuilder` and cloning the component, sound good to me. I actually originally started out cloning the component but I figured I should try to reduce duplication as the only thing that I had to change was the `matchOptions` and the `opOptions`.

I am on the fence, perhaps leaning slightly towards reducing duplication. I feel that I do not have enough experience with Kiali or with Iter8 to make an informed decision.

Do you feel strongly about either option? If it is easier, I can create two PRs, my original version with duplication and another version that builds off of this PR, and perhaps by doing so, you can have a more complete view on what I am trying to do. Let me know what you think as well!

Reply:
I'd prefer a clone as a first attempt, as Iter8 and Istio have different domains and that would give you flexibility in case that you need to change something.

The idea is that the users share a similar experience but from implementation side, it's reasonable to allow a little bit duplication on this case.

Also, I think it's easier to refactor in the future joining a cloned class that the other direction, as when you progress you'll need to check potential regressions in the Istio form.

Reply:
Okay! We can do that instead. I need to first clarify some edge case behavior with Yew Huey first, however. She said that she will get back to me next week.